File: Data Structures & Algorithms/validate-parentheses/submission-2.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> storage;
        for(const auto& entry : s) {
            if(entry == '(' || entry == '{' || entry == '[') storage.push(entry);
            else {
                if (storage.size() == 0) return false;
                else if (entry == ')' && storage.top() != '(') return false;
                else if (entry == '}' && storage.top() != '{') return false;
                else if (entry == ']' && storage.top() != '[') return false;
                else storage.pop();
            }
        }
        if (storage.size() != 0) return false;
        else return true;
    }
};

File: Data Structures & Algorithms/simplify-path/submission-2.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        pathParts = path.split("/")
        stack = []
        for part in pathParts:
            if part == "" or part == ".":
                continue
            if part == "..":
                if stack: stack.pop()
            else:
                stack.append("/"+part)
        if stack:
            answer = "".join(stack)
            return answer
        else:
            return "/"